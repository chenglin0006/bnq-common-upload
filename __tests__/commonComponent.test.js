import {shallow,mount} from 'enzyme';
import React from 'react';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import CommonComponent from '../src/index'
Enzyme.configure({ adapter: new Adapter() });
import renderer from 'react-test-renderer';

test('filter', () => {
    const props = {
        // Jest 提供的mock 函数
        testClick:jest.fn( (e) => {
            console.log('test','=============')
        }),
    }

    const component = renderer.create(
        <CommonComponent {...props}>Facebook</CommonComponent>,
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    const filterWrapperTest = mount(<CommonComponent {...props} />);
    filterWrapperTest.find('button.test').simulate('click');

});
