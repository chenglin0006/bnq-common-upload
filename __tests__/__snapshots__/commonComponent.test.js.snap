// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`filter 1`] = `
<div
  className="clearfix define-upload-wall pic-list-div test"
>
  <span>
    描述图片
    <span
      style={
        Object {
          "color": "red",
        }
      }
    >
      （
      图片说明
      ）
    </span>
  </span>
  <span
    className=""
  >
    <div
      className="ant-upload ant-upload-select ant-upload-select-"
      style={
        Object {
          "display": "",
        }
      }
    >
      <span
        className="ant-upload"
        onClick={[Function]}
        onDragOver={[Function]}
        onDrop={[Function]}
        onKeyDown={[Function]}
        role="button"
        tabIndex="0"
      >
        <input
          accept=""
          multiple={true}
          onChange={[Function]}
          style={
            Object {
              "display": "none",
            }
          }
          type="file"
        />
        <button
          className="ant-btn"
          onClick={[Function]}
          type="button"
        >
          <i
            className="anticon anticon-upload"
          />
          <span>
             点击上传
          </span>
        </button>
      </span>
    </div>
    <div
      className="ant-upload-list ant-upload-list-"
    >
      <div
        className="ant-upload-list-item ant-upload-list-item-undefined"
      >
        <div
          className="ant-upload-list-item-info"
        >
          <span>
            <i
              className="anticon anticon-paper-clip"
            />
            <a
              className="ant-upload-list-item-name"
              href="http://res1.bnq.com.cn/003f1937-60b9-440b-b29b-4e152e235a25?t=1537261779453"
              onClick={[Function]}
              rel="noopener noreferrer"
              target="_blank"
            />
          </span>
        </div>
        <i
          className="anticon anticon-cross"
          onClick={[Function]}
          title="Remove file"
        />
      </div>
      <div
        className="ant-upload-list-item ant-upload-list-item-undefined"
      >
        <div
          className="ant-upload-list-item-info"
        >
          <span>
            <i
              className="anticon anticon-paper-clip"
            />
            <a
              className="ant-upload-list-item-name"
              href="http://res1.bnq.com.cn/d1d85701-e7ff-471f-89d2-4a23960845ec?t=1537264755200"
              onClick={[Function]}
              rel="noopener noreferrer"
              target="_blank"
            />
          </span>
        </div>
        <i
          className="anticon anticon-cross"
          onClick={[Function]}
          title="Remove file"
        />
      </div>
    </div>
  </span>
  
  <div
    className="img-list-div"
  >
    <div
      className="img-item"
    >
      <img
        onClick={[Function]}
        src="http://res1.bnq.com.cn/003f1937-60b9-440b-b29b-4e152e235a25?t=1537261779453"
        style={
          Object {
            "cursor": "pointer",
          }
        }
      />
      <div
        className="buttons-div"
      >
        <button
          className="ant-btn ant-btn-sm"
          onClick={[Function]}
          type="button"
        >
          <span>
            前 移
          </span>
        </button>
        <button
          className="ant-btn ant-btn-sm"
          onClick={[Function]}
          type="button"
        >
          <span>
            后 移
          </span>
        </button>
        <button
          className="ant-btn ant-btn-sm"
          onClick={[Function]}
          type="button"
        >
          <span>
            删 除
          </span>
        </button>
      </div>
    </div>
    <div
      className="img-item"
    >
      <img
        onClick={[Function]}
        src="http://res1.bnq.com.cn/d1d85701-e7ff-471f-89d2-4a23960845ec?t=1537264755200"
        style={
          Object {
            "cursor": "pointer",
          }
        }
      />
      <div
        className="buttons-div"
      >
        <button
          className="ant-btn ant-btn-sm"
          onClick={[Function]}
          type="button"
        >
          <span>
            前 移
          </span>
        </button>
        <button
          className="ant-btn ant-btn-sm"
          onClick={[Function]}
          type="button"
        >
          <span>
            后 移
          </span>
        </button>
        <button
          className="ant-btn ant-btn-sm"
          onClick={[Function]}
          type="button"
        >
          <span>
            删 除
          </span>
        </button>
      </div>
    </div>
  </div>
</div>
`;
